src/tree.rs

```diff
@@ -827,6 +1005,87 @@ enum Divergence {
     Diverged,
 }
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum Problem {
+    /// The item doesn't have a `parentid`, and isn't mentioned in any
+    /// folder's `children`.
+    Orphan,
+
+    /// The item is a user content root (menu, mobile, toolbar, or unfiled),
+    /// but its parent isn't the Places root.
+    MisparentedRoot(Guid),
+
+    /// Multiple folders reference the item in their `children`.
+    MultipleParents(Guid),
+
+    /// The item's `parentid` doesn't match that folder's `children`.
+    ParentChildDisagreement(Guid),
+
+    /// The item's `parentid` doesn't exist.
+    MissingParentGuid(Guid),
```

Comment:
This is like `orphans` and `deletedParents`—we don't record deleted items separately from nonexistent ones, but maybe we should?

Reply:
Maybe just `MissingParent` - the `Guid` on the tail made me thing it meant the *guid* was missing in the child, but IIUC, this means the item identified by the guid doesn't exist (and that if the guid was missing it would just be `Orphan`)? Or maybe I'm missing something?n